### PR TITLE
Add extra docs to on_initialize

### PR DESCRIPTION
**Pull request by tomusdrw**
During some internal discussions it turned out it's unclear that `on_initialize` is called before all extrinsics (including inherent extrinsics), this PR attempts to add some extra docs to emphasize that.
Note there is already a doc on the trait level, but I chose to add it on the function level as well.

### CODE REVIEW

File: frame/support/src/traits.rs [hunk 1]
```diff
@@ -1438,6 +1441,9 @@ pub trait OnInitialize<BlockNumber> {
 	/// The block is being initialized. Implement to have something happen.
 	///
 	/// Return the non-negotiable weight consumed in the block.
+	/// NOTE: This function is called BEFORE ANY extrinsic in a block is applied,
+	/// including inherent extrinsics. Hence for instance `timestamp`
+	/// is not updated at this point yet.
```
Comment (dvdplm) [inline frame/support/src/traits.rs:1446]:
```suggestion
	/// is not yet up to date at this point.
```

**Review by dvdplm — APPROVED** [review]

File: frame/support/src/traits.rs [hunk 2]
```diff
@@ -1438,6 +1441,9 @@ pub trait OnInitialize<BlockNumber> {
 	/// The block is being initialized. Implement to have something happen.
 	///
 	/// Return the non-negotiable weight consumed in the block.
+	/// NOTE: This function is called BEFORE ANY extrinsic in a block is applied,
```
Comment (bkchr) [inline frame/support/src/traits.rs:1444]:
```suggestion
	///
	/// NOTE: This function is called BEFORE ANY extrinsic in a block is applied,
```

File: frame/support/src/traits.rs [hunk 3]
```diff
@@ -1438,6 +1441,9 @@ pub trait OnInitialize<BlockNumber> {
...
+	/// including inherent extrinsics. Hence for instance `timestamp`
```
Comment (bkchr) [inline frame/support/src/traits.rs:1445]: We should not mention timestamp here. Timestamp is not a required pallet of a substrate based runtime

**Review by bkchr — APPROVED** [review]

File: frame/support/src/traits.rs (hunk 3 above)
Comment (tomusdrw) [reply frame/support/src/traits.rs]: I've rephrased the comment to make it conditional. I'd rather have a concrete example here and I suppose that `timestamp` is rather a common component in the runtime and highly likely to be a common mistake as well.

Fine with removing it as well if you don't like the new phrasing either.

**Pull request merged**